meilisearch-http/src/helpers/normalize_slashes.rs

```diff
@@ -0,0 +1,85 @@
+///
+/// This middleware normalizes slashes in paths
+/// * consecutive instances of `/` get collapsed into one `/`
+/// * any ending `/` is removed.
+///
+/// Ex:
+///   /this///url/
+///   becomes : /this/url
+///
+use actix_service::{Service, Transform};
+use actix_web::{
+    dev::ServiceRequest,
+    dev::ServiceResponse,
+    http::uri::{PathAndQuery, Uri},
+    Error as ActixError,
+};
+use futures::future::{ok, Ready};
+use regex::Regex;
+use std::task::{Context, Poll};
+
+pub struct NormalizeSlashes;
+
+impl<S, B> Transform<S> for NormalizeSlashes
+where
+    S: Service<Request = ServiceRequest, Response = ServiceResponse<B>, Error = ActixError>,
+    S::Future: 'static,
+{
+    type Request = ServiceRequest;
+    type Response = ServiceResponse<B>;
+    type Error = ActixError;
+    type InitError = ();
+    type Transform = SlashNormalization<S>;
+    type Future = Ready<Result<Self::Transform, Self::InitError>>;
+
+    fn new_transform(&self, service: S) -> Self::Future {
+        ok(SlashNormalization { service })
+    }
+}
+
+pub struct SlashNormalization<S> {
+    service: S,
+}
+
+impl<S, B> Service for SlashNormalization<S>
+where
+    S: Service<Request = ServiceRequest, Response = ServiceResponse<B>, Error = ActixError>,
+    S::Future: 'static,
+{
+    type Request = ServiceRequest;
+    type Response = ServiceResponse<B>;
+    type Error = ActixError;
+    type Future = S::Future;
+
+    fn poll_ready(&mut self, cx: &mut Context) -> Poll<Result<(), Self::Error>> {
+        self.service.poll_ready(cx)
+    }
+
+    fn call(&mut self, mut req: ServiceRequest) -> Self::Future {
+        let head = req.head();
+
+        let path = head.uri.path();
+        let original_len = path.len();
+        let slash_regex = Regex::new("//+").unwrap();
+        let new_path = slash_regex.replace_all(path, "/");
+        let new_path = new_path.trim_end_matches("/");
```

Comment:
Is it where the url trailing slashes are removed?

Reply:
it removes multiples slash and trailing slash.